- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing…

rbleattler opened the pull request:
# PR Summary

- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
- Fixed bad link in README.md

## PR Checklist

- [X] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [X] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [X] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#update-changelogmd)

Commit:
- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
- Fixed bad link in README.md

SebastianSchuetze commented on CHANGELOG.md:
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
```
```suggestion
## 7.8.0

- Added Update-VSTeamGitRepositoryDefaultBranch to allow for changing the default branch of a repository.
```

SebastianSchuetze replied:
For a new freature you can raise the minor version. Can you please do the same in the PS manifest?

SebastianSchuetze commented on Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1:
```diff
@@ -0,0 +1,61 @@
+         # Storing the object before you return it cleaned up the pipeline.
+         # When I just write the object from the constructor each property
+         # seemed to be written
+         # $repo = [vsteam_lib.GitRepository]::new($resp, $ProjectName)
```
Could you explain this to me?

rbleattler replied:
This was a remnant from the function I used as a basis for creating this. Will remove and re-commit.

SebastianSchuetze commented on Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1:
```diff
@@ -0,0 +1,62 @@
+      #          $Method -eq 'Patch' -and $Uri -eq "https://localhost:8080/tfs/PeopleTracker/PeopleTracker/_apis/git/repositories/00000000-0000-0000-0000-000000000000?api-version=$(_getApiVersion Git)"
+      #       }
+      #    }
+      # }
```
Please remove it if not needed. Or are you not finished yet?

Or do you try to solve something specific?

rbleattler replied:
I was trying to follow convention used in a different test but am not particularly proficient with pester and wanted to get this out there. Apologies for not removing it.

rbleattler replied:
Cleaning up, and re-committing.

SebastianSchuetze commented on Tests/function/tests/Update-VSTeamGitRepositoryDefaultBranch.Tests.ps1:
```diff
@@ -0,0 +1,62 @@
+
+            ## Assert
+            Should -Invoke Invoke-RestMethod -ParameterFilter {
+               Write-Debug "Method : $Method"
```
Mhh this works? Never thought of that!

rbleattler replied:
The debug lines (meant to remove, again sorry -- working on this _at work_ while working on some migration stuff so in a time crunch) or the assertion itself?

SebastianSchuetze commented on Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1:
```diff
@@ -0,0 +1,61 @@
+# area            : git
+# resourceName    : repositories
+# routeTemplate   : {project}/_apis/{area}/{resource}/{repositoryId}
+# http://bit.ly/Add-VSTeamGitRepository
```
```suggestion
# http://bit.ly/Add-VSTeamGitRepository
```
Please update this to the same name as the cmdlet

SebastianSchuetze commented on Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1:
```diff
@@ -0,0 +1,61 @@
+   )
+   begin {
+      try {
+         # $Repo = Get-VSTeamGitRepository -Name $Name -ProjectName $ProjectName
```
Why are you not using this cmdlet? We are fine to use existing cmdlet to make more complex ones.

rbleattler replied:
Honestly, I believe I intended to be revert this once the tests were working, but this is related to the aforementioned issue with Pester.

rbleattler replied:
Fixed, re-committing.

SebastianSchuetze commented on Source/Public/Update-VSTeamGitRepositoryDefaultBranch.ps1:
```diff
@@ -0,0 +1,61 @@
+      
+      
+      $body = @{  
+         # name          = $Name
```
Also remove as soon as it is clear what you wanna do.

rbleattler replied:
Cleaning up, and re-committing.

SebastianSchuetze reviewed (changes requested):
Hey @rbleattler thank you for the contribution!

After reviewing it feels like you are not completely sure about your approach with some parts of the code. We need to clarify and clean this up before we can merge it!

Commit:
Clean up comments, and fixed/updated urls

Commit:
Added logic to only try to import vstem-lib.dll if exists and is not imported.

Commit:
Added fake Get-VSTeamGitRepository call (since pester doesn't recognize it as something when mocked)
Removed call to dot source common.ps1
Removed debug messages
Cleaned up comments
Bumped Version
Updated Changelog

rbleattler commented:
Closing and re-opening with proposed changes.
